optimum/neuron/pipelines/diffusers/pipeline_utils.py

```diff
@@ -24,7 +24,11 @@
 
 class DiffusionBasePipelineMixin:
     def check_num_images_per_prompt(self, prompt_batch_size: int, neuron_batch_size: int, num_images_per_prompt: int):
-        if not self.dynamic_batch_size and neuron_batch_size != prompt_batch_size * num_images_per_prompt:
+        if (
+            not self.data_parallel_mode == "all"
```

Comment:
shall we do a separate calculation check for when the mode is "all" instead of just letting it pass unconditionally when it's "all"?

Reply:
Currently there are some constraints on compiling LCM with bs>1. I would leave a placeholder here and do the check better once I sort the compilation issue out with the Annapurna team.
(for now LCM is the only case where we need to load the "all" pipeline into both cores)